submarine-server/server-core/src/main/java/org/apache/submarine/server/workbench/rest/MetricRestApi.java

```diff
@@ -0,0 +1,183 @@
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+@Path("/metric")
+@Produces("application/json")
+@Singleton
+public class MetricRestApi {
+  private static final Logger LOG = LoggerFactory.getLogger(LoginRestApi.class);
+  MetricService metricService = new MetricService();
+
+  @Inject
+  public MetricRestApi() {
+  }
+
+  /*
+  # +-------+----------+--------------+---------------+------+--------+------------------+
+  # | key   | value    | worker_index | timestamp     | step | is_nan | job_name         |
+  # +-------+----------+--------------+---------------+------+--------+------------------+
+  # | score | 0.666667 | worker-1     | 1569139525097 |    0 |      0 | application_1234 |
+  # | score | 0.666667 | worker-1     | 1569149139731 |    0 |      0 | application_1234 |
+  # | score | 0.666667 | worker-1     | 1569169376482 |    0 |      0 | application_1234 |
+  # | score | 0.666667 | worker-1     | 1569236290721 |    0 |      0 | application_1234 |
+  # | score | 0.666667 | worker-1     | 1569236466722 |    0 |      0 | application_1234 |
+  # +-------+----------+--------------+---------------+------+--------+------------------+
+  */
+
+  @GET
+  @Path("/list")
+  @SubmarineApi
+  public Response listMetric(@QueryParam("metricKey") String metricKey,
+                              @QueryParam("value") Float value,
+                              @QueryParam("workerIndex") String workerIndex,
+                              @QueryParam("timestamp") BigInteger timestamp,
+                              @QueryParam("step") Integer step,
+                              @QueryParam("isNan") Integer isNan,
+                              @QueryParam("jobName") String jobName,
+                              @QueryParam("id") String id) {
+
+    Metric metric = new Metric();
+    metric.setMetricKey(metricKey);
+    metric.setValue(value);
+    metric.setWorkerIndex(workerIndex);
+    metric.setTimestamp(timestamp);
+    metric.setStep(step);
+    metric.setIsNan(isNan);
+    metric.setJobName(jobName);
+    metric.setId(id);
+
+    LOG.info("listMetric ({})", metric);
+
+    List<Metric> metrics;
+    try {
+      metrics = metricService.selectByPrimaryKeySelective(metric);
+
+    } catch (Exception e) {
+
+      LOG.error(e.toString());
+      e.printStackTrace();
+      return new JsonResponse.Builder<String>(Response.Status.OK).success(true).result("error").build();
+    }
+    return new JsonResponse.Builder<List<Metric>>(Response.Status.OK).success(true).result(metrics).build();
+  }
+
+  @GET
+  @Path("/")
+  @SubmarineApi
+  public Response getMetric(@QueryParam("id") String id) {
+    Metric metric;
+    try {
+      metric = metricService.selectById(id);
+
+    } catch (Exception e) {
+
+      LOG.error(e.toString());
+      e.printStackTrace();
+      return new JsonResponse.Builder<String>(Response.Status.OK).success(true).result("error").build();
```

Comment:
`Response.Status.OK` means the rest request status was successful.
`success(true)` or `success(flse)` indicates success or failure of the operation.
So, Here should be changed to
```
return new JsonResponse.Builder<String>(Response.Status.OK).success(false).build();
```

Other places in this PR also need to be revised.